image/manifest_list.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/opencontainers/image-spec/schema"
+	"github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type manifest_list struct {
+	Manifests []ManifestDescriptor `json:"manifests"`
+}
+
+type ManifestDescriptor struct {
+	Descriptor v1.Descriptor
+
+	// Platform describes the platform which the image in the manifest runs on.
+	Platform Platform `json:"platform"`
+}
+
+func findManifest_list(w walker, d *descriptor) (*manifest_list, error) {
```

Comment:
Mixing camelCase and underscores is strange.  This should probably be `findManifestList`, although now that we have #29's `(r *tarReader) Get` and will hopefully soon have #5's CAS engine, I don't see much point to the `find*` functions.